src/sinks/pulsar/config.rs

```diff
@@ -206,6 +210,25 @@ pub struct CustomConnectionRetryOptions {
     pub keep_alive_secs: Option<u64>,
 }
 
+#[configurable_component]
+#[configurable(description = "TLS options configuration for the Pulsar client.")]
+#[derive(Clone, Debug)]
+pub struct PulsarTlsOptions {
```

Comment:
Would it be possible to reuse the existing config struct we use for TLS options: https://github.com/vectordotdev/vector/blob/0720345ef61398819174290fd6c4de6d0e1e9446/lib/vector-core/src/tls/settings.rs#L87?

Or does the pulsar client no support all of the same options?

At the least, we should match the option names (e.g. `certificate_chain_file` should be `ca_file` to match).

Reply:
Thanks for your feedback. 
rust client for pulsar only supports some options https://github.com/streamnative/pulsar-rs/blob/master/src/connection_manager.rs#L73
Changed to use the same field names as TlsConfig [66821d8](https://github.com/vectordotdev/vector/pull/22148/commits/66821d880245535f0611176b68991c9cff809533).